packages/headless/src/controllers/commerce/context/cart/headless-cart-selectors.ts

```diff
@@ -1,9 +1,12 @@
 import {createSelector} from '@reduxjs/toolkit';
 import {itemsSelector} from '../../../../features/commerce/context/cart/cart-selector';
 import {CartState} from '../../../../features/commerce/context/cart/cart-state';
+import {CartItem, createCartKey} from './headless-cart';
 
-export function itemSelector(cartState: CartState, sku: string) {
-  return cartState.cart[sku];
+export function itemSelector(cartState: CartState, cartItem: CartItem) {
+  return Object.entries(cartState.cart[createCartKey(cartItem)])
+    .filter(([_, item]) => item.price === cartItem.price)
+    .map(([_, item]) => item)[0];
```

Comment:
I think by adding the stringify price to the key, it would simplify the logic here.

```suggestion
  return cartState.cart[createCartKey(cartItem)]
```